ff/serving/metadata/metadata.go

```diff
@@ -775,6 +793,47 @@ func NewMetadataServer(logger *zap.SugaredLogger) (*MetadataServer, error) {
 	}, nil
 }
 
+func NewMetadataTypesenseServer(logger *zap.SugaredLogger, host string, port string, apiKey string) (*MetadataServer, error) {
+	logger.Debug("Creating new metadata server")
+	client := typesense.NewClient(
+		typesense.WithServer(fmt.Sprintf("http://%s:%s", host, port)),
+		typesense.WithAPIKey(apiKey))
+	logger.Debugf("Creating typsense client on http://%s:%s with apiKey %s", host, port, apiKey)
+	schema := &api.CollectionSchema{
+		Name: "resource",
+		Fields: []api.Field{
+			{
+				Name: "Name",
+				Type: "string",
+			},
+			{
+				Name: "Variant",
+				Type: "string",
+			},
+			{
+				Name: "Type",
+				Type: "string",
+			},
+		},
+	}
+	client.Collections().Create(schema)
+	var resourceinitial []interface{}
+	action := "create"
+	batchnum := 40
+	params := &api.ImportDocumentsParams{
+		Action:    &action,
+		BatchSize: &batchnum,
+	}
+	client.Collection("resource").Documents().Import(resourceinitial, params)
```

Comment:
Why are you importing documents?

Reply:
to initialize the collection I imported an empty interface so we could just call the Upsert function in set

Reply:
yes it would throw an error

Reply:
Could you add a comment above this explaining that since it's non-obvious